web/src/server_events_dispatch.js

```diff
@@ -1198,6 +1195,9 @@ export function dispatch_normal_event(event) {
                     user_status.get_status_emoji(event.user_id),
                 );
             }
+            if (event.user_id === current_user.user_id) {
+                user_status.set_scheduled_end_time(event.scheduled_end_time);
```

Comment:
I think this will set `scheduled_end_time` to undefined if it is not passed in the event when it is not updated. So, would be better to have a check for undefined value as well.